src/dev/build/tasks/os_packages/docker_generator/run.js

```diff
@@ -0,0 +1,101 @@
+ * under the License.
+ */
+
+import { access, link, unlink } from 'fs';
+import { resolve } from 'path';
+import { promisify } from 'util';
+import { write, copyAll, exec, mkdirp } from '../../../lib';
+import * as dockerTemplates from './templates';
+
+const accessAsync = promisify(access);
+const linkAsync = promisify(link);
+const unlinkAsync = promisify(unlink);
+
+export async function runDockerGenerator(config, log, build) {
+  const license = build.isOss() ? 'ASL 2.0' : 'Elastic License';
+  const imageFlavor = build.isOss() ? '-oss' : '';
+  const imageTag = 'docker.elastic.co/kibana/kibana';
+  const versionTag = config.getBuildVersion();
+  const artifactTarball = `kibana${ imageFlavor }-${ versionTag }-linux-x86_64.tar.gz`;
+  const artifactsDir = config.resolveFromTarget('.');
+  const dockerBuildDir = config.resolveFromRepo('build', 'kibana-docker', build.isOss() ? 'oss' : 'default');
+  const dockerOutputDir = config.resolveFromTarget(`kibana${ imageFlavor }-${ versionTag }-docker.tar`);
+
+  // Verify if we have the needed kibana target in order
+  // to build the kibana docker image.
+  // Also create the docker build target folder
+  // and  delete the current linked target into the
+  // kibana docker build folder if we have one.
+  try {
+    await accessAsync(resolve(artifactsDir, artifactTarball));
+    await mkdirp(dockerBuildDir);
+    await unlinkAsync(resolve(dockerBuildDir, artifactTarball));
+  } catch (e) {
+    if (e && e.code === 'ENOENT' && e.syscall === 'access') {
+      throw new Error(
+        `Kibana linux target (${ artifactTarball }) is needed in order to build ${''
+        }the docker image. None was found at ${ artifactsDir }`
+      );
+    }
+  }
+
+  // Create the kibana linux target inside the
+  // Kibana docker build
+  await linkAsync(
+    resolve(artifactsDir, artifactTarball),
+    resolve(dockerBuildDir, artifactTarball),
+  );
+
+  // Write all the needed docker config files
+  // into kibana-docker folder
+  const scope = {
+    artifactTarball,
+    imageFlavor,
+    versionTag,
+    license,
+    artifactsDir,
+    imageTag,
+    dockerOutputDir
+  };
+
+  for (const [, dockerTemplate] of Object.entries(dockerTemplates)) {
+    await write(resolve(dockerBuildDir, dockerTemplate.name), dockerTemplate.generator(scope));
+  }
+
+  // Copy all the needed resources into kibana-docker folder
+  // in order to build the docker image accordingly the dockerfile defined
+  // under templates/kibana_yml.template/js
+  await copyAll(
+    config.resolveFromRepo('src/dev/build/tasks/os_packages/docker_generator/resources'),
+    dockerBuildDir,
+  );
+
+  // Build docker image into the target folder
+  // In order to do this we just call the file we
+  // created from the templates/build_docker_sh.template.js
+  // and we just run that bash script
+  const args = [
+    `${resolve(dockerBuildDir, 'build_docker.sh')}`
+  ];
+
+  await exec(log, `sh`, args, {
```

Comment:
Instead of invoking `sh` here directly, can we instead call the file and allow env to work it's magic?

```
await exec(log, './build_docker.sh', [], {
  cwd: dockerBuildDir,
  level: 'info',
});
```

Reply:
@tylersmalley I choose to not do it like this as I was having an error:

```
$ node scripts/build --all-platforms --docker
 info [  kibana  ] Creating docker package
   │ info $ ./build_docker.sh
   │ERROR failure
   │ERROR TypeError: Cannot read property 'on' of undefined
   │          at destroyer (internal/streams/pipeline.js:27:10)
   │          at internal/streams/pipeline.js:79:12
   │          at Array.map (<anonymous>)
   │          at pipeline (internal/streams/pipeline.js:76:28)
   │          at Promise (/Users/tiagocosta/elastic/kibana/src/utils/streams/promise_from_streams.js:59:17)
   │          at new Promise (<anonymous>)
   │          at createPromiseFromStreams (/Users/tiagocosta/elastic/kibana/src/utils/streams/promise_from_streams.js:58:10)
   │          at watchStdioForLine (/Users/tiagocosta/elastic/kibana/src/utils/watch_stdio_for_line.js:66:5)
   │          at exec (/Users/tiagocosta/elastic/kibana/src/dev/build/lib/exec.js:41:9)
   │          at runDockerGenerator (/Users/tiagocosta/elastic/kibana/src/dev/build/tasks/os_packages/docker_generator/run.js:97:9)
   │          at process._tickCallback (internal/process/next_tick.js:68:7)
```

Reply:
@tylersmalley just solve this and I'm now using the call the file relatively!